packages/ui/src/utilities/buildFormState.ts

```diff
@@ -207,5 +230,82 @@ export const buildFormState = async ({ req }: { req: PayloadRequest }): Promise<
     }
   }
 
-  return result
+  if (returnLockStatus && req.user && (id || globalSlug)) {
+    let lockedDocumentQuery
+
+    if (collectionSlug) {
+      lockedDocumentQuery = {
+        and: [
+          { 'document.relationTo': { equals: collectionSlug } },
+          { 'document.value': { equals: id } },
+        ],
+      }
+    } else if (globalSlug) {
+      lockedDocumentQuery = {
+        globalSlug: { equals: globalSlug },
+      }
+    }
+
+    if (lockedDocumentQuery) {
```

Comment:
This should really be it's own function that returns the lockStatus. Can always refactor later.